framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -283,6 +284,12 @@ public void updateFromFieldAssignment(
                             + lhs.getClass());
         }
 
+        // Do not attempt to infer types for temporary variables.
+        // See https://github.com/typetools/checker-framework/issues/3442
+        if (!SourceVersion.isIdentifier(fieldName)) {
```

Comment:
Two questions.

1. Is it having a valid identifier equivalent equivalent to being a temporary variable? If not, then this fix might work for the issue in question, but still do inference on other temporary nodes. I noticed near [this](https://github.com/typetools/checker-framework/blob/d3da57a5f9e0bc3b9bbf294159dc7283ec0a8c16/dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java#L4759) line from the code that creates the variable that `setInSource(false)` is called on the generated nodes. Would something like that work here instead?

2. Why is the temporary variable considered a field in the first place? I feel like we *would* want to do inference on temporary variables in an expression like `int n = i++` because the temporary variable would hold the inferred type of the expression `i++`. It seems to me like the root of the problem is that in static blocks in particular, the temporary variable is somehow "attached" to the class as a field.

Reply:
1. I decided to check the identifier because I think it's broader than checking whether the tree was actually generated by the compiler (i.e. `setInSource` being set to `false`). My reasoning is that because we're going to write the results of inference out to a jaif/stub file (neither of which can include fields with invalid names), we should handle ANY invalid names that we encounter, regardless of where they came from.

2. 
> Why is the temporary variable considered a field in the first place
I think it's just a quirk of how the CFG is constructed.
> I feel like we would want to do inference on temporary variables in an expression like `int n = i++`
We'll still do WPI on `n` in this example, and we'll compute the correct results because *local* inference will still run on the `i++` node - but only `n`'s inferred type will be written to the jaif/stub file.

> It seems to me like the root of the problem is that in static blocks in particular, the temporary variable is somehow "attached" to the class as a field
Yes. I presume there's a reason that was done. If there isn't (and it's just an implementation detail that could be changed), I would still prefer this fix, because it has a much smaller blast radius (only WPI is impacted). Changing CFG construction could result in bugs elsewhere.